parse_test.go

```diff
@@ -45,6 +45,30 @@ func TestParse(t *testing.T) {
 	}
 }
 
+// TestNewParserSkipHeaderRead tests that NewParser with the SkipHeaderRead option
+// parses the specified dataset but not its header.
```

Comment:
Note, in practice this way of skipping the header won't _actually_ skip the header data in the dicom if it's present (it'll just stop the parser from running the metadata read section of the code).

Maybe we should make that more clear in the option?

That is, if I pass a dicom that has header data into a Parser initialized with SkipHeaderRead, whenever we start reading elements from that parser, we will still get Metadata elements in our normal parse process. In otherwords, we don't advance the internal reader to skip metadata elements if there are any in the dicom, right?

I think maybe this could be a point of confusion for a user for an option entitled `SkipHeaderRead`. One option is that we read the metadata elements (if any) but throw them all away. Or, perhaps we should document this option more carefully--maybe call it something like `SkipMetadataReadOnNewParserInit` or something to make it clear that all the option does is prevent the NewParser() from trying to read metadata. But if metadata is actually present in the dicom, it's not like we skip parsing those elements (we should mention this in the comment regardless of what we call it).

Let me know if this makes sense! Apologies for the back and forth--the test is useful for seeing things in this framing.